### PR TITLE
Add differentiation tests

### PR DESCRIPTION
**Pull request by devmotion**
I added tests similar to the ones in https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/master/test/interface/ad_tests.jl. Unfortunately they show that AD doesn't work (as suspected in https://github.com/JuliaDiffEq/DelayDiffEq.jl/issues/133).

### CODE REVIEW

File: test/differentiation.jl [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  nothing
+end
+
+h(p, t) = [p[4]]
+
+@testset "Gradient" begin
+  function test(p)
+    prob = DDEProblem(f, [p[5]], h, eltype(p).((0.0, 20.0)), copy(p);
+                      constant_lags = [p[2]])
+    sol = solve(prob, MethodOfSteps(Tsit5()))
```
Comment (ChrisRackauckas) [inline test/differentiation.jl]: you have to set the tolerance really low to test the difference here, since finite difference is really sensitive.

Comment (devmotion) [reply test/differentiation.jl]: Ah OK. However, I guess it does not matter at the moment - the values are off by multiple orders of magnitude, and even the signs do not match.

Comment (ChrisRackauckas) [reply test/differentiation.jl]: yeah that's bad haha

**Comment by devmotion** [comment]
Tests on AppVeyor took too long to run, but the differentiation tests passed successfully.

**Pull request merged**